docs-src/cloud/terraform-provider.md

```diff
@@ -0,0 +1,164 @@
+
+The GitHub repository for the Terraform provider is [terraform-provider-temporalcloud](https://github.com/temporalio/terraform-provider-temporalcloud/tree/main), where you can find the source code, contribution guidelines, and additional information.
+
+:::note
+
+Temporal supports the [Cloud Namespace](/cloud/namespaces) resource management, a user resource for managing Temporal Cloud users, and data sources for retrieving information about existing Namespaces and users, with the Temporal Cloud Terraform provider.
+
+:::
+
+### Prerequisites
+
+You must have access to the following to use the Temporal Cloud Terraform provider:
+
+- [Terraform account](https://developer.hashicorp.com/terraform)
+- [Terraform CLI](https://developer.hashicorp.com/terraform/cli)
+- [Temporal tcld](/cloud/tcld/)
+- [Issued certification authority](/cloud/certificates)
+
+## Get started
+
+This walk you through getting started with the Temporal Cloud Terraform provider.
+You will generate an API Key used to authenticate your requests with your Temporal Cloud account.
+Then you will create a Temporal Cloud Namespace with the Terraform provider.
+
+### 1. Generate an API Key
+
+You can skip generating an API Key if you already possess a valid API Key.
+
+1. Use the tcld to log in and generate an API Key.
+   This API Key is used to authenticate the Terraform provider run.
+
+```bash
+# authenticate your session
+tcld login
+# generate an API Key
+tcld apikey create -n "terraform-test" --desc "Testing the API Key for the TF Provider" -d 90d
+```
+
+You will use this key when creating or destroying Namespaces with the Terraform provider.
+
+2. Provide your working environment with the API Key generated in the previous step.
+
+<Tabs>
+  <TabItem value="macos" label="macOS" default>
+Export your environment variable for secure access to the API Keys.
+
+```bash
+# replace <yoursecretkey> with the "secretKey": output from the tcld apikey create command
+export TEMPORAL_CLOUD_API_KEY=<yoursecretkey>
+```
+
+</TabItem>
+  <TabItem value="windows" label="Windows">
+Export your environment variable for secure access to the API Keys.
+
+```bash
+# replace <yoursecretkey> with the "secretKey": output from the tcld apikey create command
+set TEMPORAL_CLOUD_API_KEY=<yoursecretkey>
+```
+
+</TabItem>
+</Tabs>
+
+The `TEMPORAL_CLOUD_API_KEY` is used to authenticate the Terraform provider.
+
+3. Generate your CA certificate
+
+You can skip generating your CA certificate if you already possess a CA certificate connected to your Temporal Cloud account.
+
+The CA certificate allows you to authenticate and interact with your Temporal Cloud Namespace.
+You can use an existing CA cert or [create one using tcld](/cloud/certificates).
+Once you have your CA certificate be sure to add the CA `.pem` file to your working directory.
+
+```bash
+mv ca.pem test-temporal-terraform
+```
+
+### 2. Create a Temporal Cloud Namespace with Terraform
```

Comment:
Same guidance here, as well as for Validate, and Delete.

Maybe nix the numbers in the header, add a question like sentence to the top to help frame the scope of the info.